codejam-virtual-keyboard/index.js

```diff
@@ -0,0 +1,155 @@
+const keyLayoutEng = [
+	'1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'Backspace',
+	'q', 'w', 'e', 'r', 't', 'y', 'u', 'i', 'o', 'p',
+	'CapsLock', 'a', 's', 'd', 'f', 'g', 'h', 'j', 'k', 'l', 'Enter',
+	'Shift', 'z', 'x', 'c', 'v', 'b', 'n', 'm', ',', '.', '?',
+	'Space',
+];
+const keyLayoutRu = [
+	'1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'Backspace',
+	'й', 'ц', 'у', 'к', 'е', 'н', 'г', 'ш', 'щ', 'з', 'х', 'ъ',
+	'CapsLock', 'ф', 'ы', 'в', 'а', 'п', 'р', 'о', 'л', 'д', 'ж', 'э', 'Enter',
+	'Shift', 'я', 'ч', 'с', 'м', 'и', 'т', 'ь', 'б', 'ю', '.',
+	'Space',
+];
+
+const specialKeys = ['Backspace', 'CapsLock', 'Enter', 'Shift', 'Space'];
+const lastKeysInRow = ['Backspace', 'p', 'Enter', '?', 'ъ'];
+
+const keyboardNameKey = 'current-keyboard-name';
+
+let savedKeyboardName =  localStorage.getItem(keyboardNameKey);
+
+
+let capsLock = false;
+
+let currentKeyboard = savedKeyboardName === 'RU' ? keyLayoutRu : keyLayoutEng;
+
+
+function createKeyboardButton(el) {
+	// create button
+	const button = document.createElement('div');
+	button.classList.add('keyboard__key');
+
+	if (specialKeys.includes(el)) {
+		button.classList.add('keyboard__wide');
+	}
+	if (!specialKeys.includes(el)) {
+		button.innerHTML = capsLock ? el.toUpperCase() : el.toLowerCase();
+	} else {
+		button.innerHTML = el;
+	}
+	// add click functionality
+	button.addEventListener('click', () => {
+		const input = document.getElementById('textarea');
+		switch (el) {
+		case 'Backspace':
+			input.value = input.value.slice(0, -1);
+			triggerAnimation(button);
+			break;
+
+		case 'Space':
+			input.value += ' ';
+			triggerAnimation(button);
+			break;
+
+		case 'CapsLock':
+			capsLock = !capsLock;
+			document.querySelector('body')
+				.removeChild(document.querySelector('.keyboard'));
+			createKeyboard(currentKeyboard);
+			triggerAnimation(button);
+			break;
+
+		case 'Enter':
+			input.value += '\n';
+			triggerAnimation(button);
+			break;
+
+		default:
+			input.value += button.innerHTML;
+			triggerAnimation(button);
+		}
+	});
+
+	// append key
+	document.querySelector('.keyboard')
+		.appendChild(button);
+
+	// brake after line
+	if (lastKeysInRow.includes(el)) {
+		const brTag = document.createElement('br');
+		document.querySelector('.keyboard')
+			.appendChild(brTag);
+	}
+}
+
+function triggerAnimation(el) {
+	const animation1 = setInterval(changeColor, 30);
```

Comment:
It is better to make variables on top of the file with constants like 30 here, and use that constants instead on numbers.